scripts/coccinelle/unsigned_assignment.cocci

```diff
@@ -0,0 +1,40 @@
+// Find assignments to unsigned variables and add an 'U' to the value
```

Comment:
https://docs.zephyrproject.org/latest/application/coccinelle.html#proposing-new-semantic-patches

Use `///` here.

And after doing all changes run(assuming you've installed coccinelle):
`$./scripts/coccicheck --mode=patch --verbose=1 --cocci=scripts/coccinelle/unsigned_assignment.cocci`

and  you will see the message:
`Find assignments to unsigned variables and add an 'U' to the value`

Also add `Confidence` which seems HIGH since no false positives are observed.